### PR TITLE
Mailing dependency injection

### PR DESCRIPTION
**Pull request by Vicctorri**

### CODE REVIEW

File: resources/views/home/contact.blade.php [hunk 1]
```diff
@@ -30,7 +30,16 @@
...
+                      <select id="subject" class="form-control" name="subject">
+                        <option value="">Select the subject</option>
+                        <option value="education" >Education</option>
+                        <option value="computers" >Computers</option>
+                        <option value="business" >Business</option>
+                        <option value="science" >Science & Technology</option>
+                        <option value="social" >Social Sciences</option>
+                        <option value="study" >Study Aids & Test Prep</option>
+                    </select>
+                      {{-- <input type="text" value="{{ old('subject') }}"  class="form-control" placeholder="Subject" name="subject"> --}}
```
Comment (egorbwork) [inline resources/views/home/contact.blade.php]: Comment code won't be accepted.

File: app/Services/ContactUsMailer.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        function(Message $message) use ($data){ 
+            $message->subject('mesage from ' . $data['email']);
+            $message->to('library@gmail.com');
+            $message->from('no-reply@library.app', 'ViLibrary');
+
+
+        }
+
+    );
+    $this->logger->info('conttt info ');
```
Comment (egorbwork) [inline app/Services/ContactUsMailer.php]: Please fix info message, it should provide value

File: app/Http/Request/ContactRequest.php [hunk 3]
```diff
@@ -20,7 +20,7 @@ public function rules(): array
             'lastname' => ['required', 'string', 'min:2'],
             'email' => ['required', 'string', 'email'],
             'phonenumber' => ['required', 'string', 'min:10'],
-            'subject' => ['required', 'string', 'min:2'],
+            'subject' => ['required', 'string', 'min:1', 'in:education,computers,business,science,social,study'],
```
Comment (egorbwork) [inline app/Http/Request/ContactRequest.php]: why do you min:1 here?

File: .gitignore [hunk 4]
```diff
@@ -13,3 +13,4 @@ npm-debug.log
 yarn-error.log
 /.idea
 /.vscode
+_ide_helper.php
```
Comment (egorbwork) [inline .gitignore:16]: The file was not removed from git.

**Review by egorbwork — CHANGES_REQUESTED** [review]
At this state is not ready. Multiple fixes are required.

File: .gitignore (hunk 4 above)
Comment (Vicctorri) [reply .gitignore:16]: I removed _ide_helper.php

File: resources/views/home/contact.blade.php (hunk 1 above)
Comment (Vicctorri) [reply resources/views/home/contact.blade.php]: Removed

File: app/Services/ContactUsMailer.php (hunk 2 above)
Comment (Vicctorri) [reply app/Services/ContactUsMailer.php]: Changed

**Review by egorbwork — APPROVED** [review]
Minim necessary changes were applied.

**Pull request merged**